datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -767,6 +769,71 @@ private Future<Boolean> dispatchAssignmentChangeIfNeeded(String connectorType, L
     return null;
   }
 
+  private List<DatastreamTask> getRemovedTasks(List<DatastreamTask> newAssignment, List<DatastreamTask> oldAssignment) {
+    List<DatastreamTask> removedTasks = new ArrayList<>(oldAssignment);
+    removedTasks.removeAll(newAssignment);
+    return removedTasks;
+  }
+
+  private List<DatastreamTask> getAddedTasks(List<DatastreamTask> newAssignment, List<DatastreamTask> oldAssignment) {
+    List<DatastreamTask> addedTasks = new ArrayList<>(newAssignment);
+    addedTasks.removeAll(oldAssignment);
+    return addedTasks;
+  }
+
+  @VisibleForTesting
+  void maybeClaimAssignmentTokensForStoppingStreams(List<DatastreamTask> newAssignment,
```

Comment:
Add logs in both info and debug mode for debugging. If there is any issue in production in this code path, we will not be able to identify it easily.

Reply:
This is covered with tests and already has logs in the important path (i.e. when it infers what streams are potentially stopping). I don't think there's value in adding logs in the first section where it breaks down the old and new assignments per connector.